app/controllers/sitemap_controller.rb

```diff
@@ -0,0 +1,107 @@
+class SitemapController < ActionController::Metal
+
+  include AbstractController::Rendering
+  include ActionController::MimeResponds
+  include ActionController::DataStreaming
+  include ActionController::RackDelegation
+  include ActionController::Rescue
+  include ActionController::Head
+  include ActionController::Redirecting
+
```

Comment:
With metal controller you want to add:

```ruby
  # Ensure ActiveSupport::Notifications events are fired
  include ActionController::Instrumentation
```

This triggers the structured request logging.